### DISCUSSION TITLE
C# / dotnet support

### DISCUSSION THREAD

**Discussion by ManniManfred** [slint-ui/slint#499] [General]
I think a binding to C# / dotnet would be very cool? What do you think about that?

**Comment by tronical** [comment]
Yes, that would be very cool :). I don't have any experience with binding to dotnet. How easy is it for example to integrate a separate code generator tool? What are idiomatic ways of representing data models?

**Reply by ManniManfred** [reply]
For binding between rust and dotnet, this would be PInvoke (DllImport attribute) on dotnet side.
It would also be possible to wrap the existing c++ lib with a c++/cli wrapper to use the classes in dotnet, but i think the direct way fits better?
For representing data, i would suggest POCO's and get the available properties over reflection if needed.

**Comment by ogoffart** [comment]
@microhobby Actually started dotnet support:  https://github.com/microhobby/slint-dotnet  and https://github.com/slint-ui/slint/discussions/3550
